src/Symfony/Component/Validator/Constraints/BicValidator.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * @author Michael Hirschler <michael.vhirsch@gmail.com>
+ *
+ * @link https://en.wikipedia.org/wiki/ISO_9362#Structure
+ *
+ * @api
+ */
+class BicValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        $canonicalize = str_replace(' ', '', $value);
+
+        // the bic must be either 8 or 11 characters long
+        if (8 !== strlen($canonicalize) && 11 !== strlen($canonicalize)) {
```

Comment:
However, my concern may not be very pragmatic, so go with the version most readable to your eyes :smiley: